packages/backend/src/core/StateUpdateCollector.ts

```diff
@@ -77,28 +85,75 @@ export class StateUpdateCollector {
     if (rootHash !== PedersenHash(decoded.newState.positionRoot)) {
       throw new Error('State transition calculated incorrectly')
     }
-    await this.stateUpdateRepository.add({
-      stateUpdate: {
-        id,
+    await Promise.all([
+      this.stateUpdateRepository.add({
+        stateUpdate: {
+          id,
+          blockNumber,
+          factHash,
+          rootHash,
+          timestamp,
+        },
+        positions: newPositions.map(
+          ({ value, index }): PositionRecord => ({
+            positionId: index,
+            publicKey: value.publicKey,
+            balances: value.assets,
+            collateralBalance: value.collateralBalance,
+          })
+        ),
+        prices: decoded.newState.oraclePrices,
+      }),
+      this.processForcedActions(
+        decoded.forcedActions,
+        timestamp,
         blockNumber,
-        factHash,
-        rootHash,
-        timestamp: Timestamp.fromSeconds(timestamp),
-      },
-      positions: newPositions.map(
-        ({ value, index }): PositionRecord => ({
-          positionId: index,
-          publicKey: value.publicKey,
-          balances: value.assets,
-          collateralBalance: value.collateralBalance,
-        })
+        id
       ),
-      prices: decoded.newState.oraclePrices,
-    })
+    ])
+  }
+
+  async processForcedActions(
+    forcedActions: ForcedAction[],
+    timestamp: Timestamp,
+    blockNumber: number,
+    stateUpdateId: number
+  ): Promise<void> {
+    const datas = forcedActions.map((action) =>
+      action.type === 'trade'
+        ? omit(action, 'type', 'nonce')
```

Comment:
@sin6pi7 this might be needed for hash calculations - check and update if necessary